# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/naga/src/valid/mod.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/naga/src/valid/mod.rs b/third_party/rust/naga/src/valid/mod.rs
--- a/third_party/rust/naga/src/valid/mod.rs
+++ b/third_party/rust/naga/src/valid/mod.rs
@@ -53,16 +53,18 @@ bitflags::bitflags! {
     #[derive(Default)]
     #[cfg_attr(feature = "serialize", derive(serde::Serialize))]
     #[cfg_attr(feature = "deserialize", derive(serde::Deserialize))]
     pub struct Capabilities: u8 {
         /// Support for `StorageClass:PushConstant`.
         const PUSH_CONSTANT = 0x1;
         /// Float values with width = 8.
         const FLOAT64 = 0x2;
+        /// Support for `Builtin:PrimitiveIndex`.
+        const PRIMITIVE_INDEX = 0x4;
     }
 }
 
 bitflags::bitflags! {
     /// Validation flags.
     #[cfg_attr(feature = "serialize", derive(serde::Serialize))]
     #[cfg_attr(feature = "deserialize", derive(serde::Deserialize))]
     pub struct ShaderStages: u8 {
@@ -160,23 +162,25 @@ impl crate::TypeInner {
         match *self {
             Self::Scalar { .. }
             | Self::Vector { .. }
             | Self::Matrix { .. }
             | Self::Array {
                 size: crate::ArraySize::Constant(_),
                 ..
             }
+            | Self::Atomic { .. }
             | Self::Pointer { .. }
             | Self::ValuePointer { .. }
             | Self::Struct { .. } => true,
             Self::Array { .. } | Self::Image { .. } | Self::Sampler { .. } => false,
         }
     }
 
+    /// Return the `ImageDimension` for which `self` is an appropriate coordinate.
     fn image_storage_coordinates(&self) -> Option<crate::ImageDimension> {
         match *self {
             Self::Scalar {
                 kind: crate::ScalarKind::Sint,
                 ..
             } => Some(crate::ImageDimension::D1),
             Self::Vector {
                 size: crate::VectorSize::Bi,